### PR TITLE
Feature/login logic

### PR DESCRIPTION
**Pull request by spartak0**

### CODE REVIEW

File: app/src/main/java/com/example/fitness/di/DataModule.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Singleton
+    @Provides
+    fun provideFirebaseService(@ApplicationContext context: Context):FirebaseService{
+        return FirebaseService(context = context)
+    }
+
+    @Singleton
+    @Provides
+    fun provideRepositoryImpl(firebaseService: FirebaseService): FirebaseRepositoryImpl {
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/di/DataModule.kt:25]: Why do we provide implementation? I think that clean architecture way means using abstractions, not implementations

File: app/src/main/java/com/example/fitness/ui/Screen.kt [hunk 2]
```diff
@@ -5,5 +5,5 @@ sealed class Screen(val route:String){
     object LoginScreen:Screen("login_screen")
     object FirstRegScreen:Screen("first_reg_screen")
     object SecondRegScreen:Screen("second_reg_screen")
-    object BottomBarScreen:Screen("main_screen")
+    object BottomBarScreen:Screen("bottom_bar_screen")
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/Screen.kt:8]: bottom_bar is a part of screen, not a whole screen

File: app/src/main/java/com/example/fitness/ui/bottom_nav_bar/ProfileScreen.kt [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+import com.example.fitness.ui.Screen
 import com.google.firebase.auth.FirebaseAuth
 
 @Composable
-fun ProfileScreen(navController: NavController) {
-    Text(text = "Profile", modifier = Modifier.clickable { FirebaseAuth.getInstance().signOut() })
+fun ProfileScreen(navControllerBottomBar: NavController, navControllerScreens:NavController) {
+    Column() {
+        Button(onClick = {
+            FirebaseAuth.getInstance().signOut()
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/bottom_nav_bar/ProfileScreen.kt:14]: Should be inside `FirebaseService`

File: app/src/main/java/com/example/fitness/ui/login/LoginScreen.kt [hunk 4]
```diff
@@ -59,15 +58,20 @@ fun LoginScreen(navController: NavController) {
...
         Spacer(modifier = Modifier.size(350.dp))
         GradientView(
-            text = "Login",
+            text = stringResource(id = R.string.login),
             modifier = Modifier
                 .padding(horizontal = 30.dp)
                 .height(dimensionResource(id = R.dimen.view_height))
                 .fillMaxWidth()
                 .clickable {
+                    runBlocking {
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/login/LoginScreen.kt:67]: Why we can't use `viewModelScope` inside `ViewModel`?

File: app/src/main/java/com/example/fitness/ui/main/MainActivity.kt [hunk 5]
```diff
@@ -36,20 +42,25 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun Navigation() {
     val navController = rememberNavController()
-    NavHost(navController = navController, startDestination = Screen.LoginScreen.route) {
+    val user = Firebase.auth.currentUser
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/main/MainActivity.kt:45]: Should be inside `FirebaseService`

File: app/src/main/java/com/example/fitness/ui/secondReg/SecondRegScreen.kt [hunk 6]
```diff
@@ -86,63 +87,27 @@ fun SecondRegScreen(navController: NavController, user: User) {
                 .height(dimensionResource(id = R.dimen.view_height))
                 .padding(horizontal = 30.dp)
                 .clickable {
-                    validation =
-                        secondValidationTest(user.gender, user.birthDay, user.weight, user.height)
-                    if (validation.first) {
-                        reg(user, navController.context)
+                    viewModel.validationTest(
+                        user.gender,
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/secondReg/SecondRegScreen.kt:91]: Why we can't provide whole `User` object to the method?

File: app/src/main/java/com/example/fitness/ui/secondReg/SecondRegScreen.kt [hunk 7]
```diff
@@ -181,10 +146,12 @@ fun Anthropometry(userAnthropometry: UserAnthropometry, user: User) {
         }
         when (userAnthropometry) {
             UserAnthropometry.Weight -> {
-                user.weight = if (value.isEmpty() or !value.matches("[\\d]+".toRegex())) 0 else value.toInt()
+                user.weight =
+                    if (value.isEmpty() or !value.matches("[\\d]+".toRegex())) 0 else value.toInt()
```
Comment (YellowShark) [inline app/src/main/java/com/example/fitness/ui/secondReg/SecondRegScreen.kt:150]: Can you extract that into extension function `isNum()`?

**Review by YellowShark — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fitness/ui/bottom_nav_bar/ProfileScreen.kt (hunk 3 above)
Comment (spartak0) [reply app/src/main/java/com/example/fitness/ui/bottom_nav_bar/ProfileScreen.kt:14]: This is a temporary solution to test the login. I didn't work on this screen.

**Pull request merged**